cpp/include/ray/api/object_ref.h

```diff
@@ -46,6 +47,17 @@ ObjectRef<T>::ObjectRef() {}
 template <typename T>
 ObjectRef<T>::ObjectRef(const ObjectID &id) {
   id_ = id;
+  if (CoreWorkerProcess::IsInitialized()) {
+    auto &core_worker = CoreWorkerProcess::GetCoreWorker();
+    core_worker.AddLocalReference(id_);
```

Comment:
The best way to implement this is to achieve different logic in runtime with object_store. Because we have two runtime implementations: local_mode and native. This two should be different anyway. But I think we can merge this PR firstly.